firecrown/ccl/sources/sources.py

```diff
@@ -374,3 +374,80 @@ def render(self, cosmo, params, systematics=None):
             systematics[systematic].apply(cosmo, params, self)
 
         self.dndz_interp_ = Akima1DInterpolator(self.z_, self.dndz_)
+
+
+class CMBLSource(Source):
+    """A CCL CMB Lensing Source.
+
+    Parameters
+    ----------
+    sacc_tracer : str
+        The name of the tracer in the SACC file.
+    scale : float, optional
+        The default scale for this source. Usually the default of 1.0 is
+        correct.
+    systematics : list of str, optional
+        A list of the source-level systematics to apply to the source. The
+        default of `None` implies no systematics.
+
+    Attributes
+    ----------
+    beam_orig : np.ndarray
+        The original beam used in the sacc tracer before any systematics
+        are applied. Set after a call to `read`.
+    beam_: np.ndarray
+        The beam used for CMB lensing. Set after a call to `render`.
+    scale_ : float
+        The overall scale associated with the source. Set after a call to
+        `render`.
+    tracer_ : `pyccl.WeakLensingTracer`
+        The CCL tracer associated with this source. Set after a call to
+        `render`.
+
+    Methods
+    -------
+    render : apply systematics to this source and build the
+        `pyccl.CMBLSource`
+    """
+    def __init__(self, *, sacc_tracer, scale=1.0, systematics=None):
+        self.sacc_tracer = sacc_tracer
+        self.scale = scale
+        self.systematics = systematics or []
+
+    def read(self, sacc_data):
+        """
+        Read the data for this source from the SACC file.
+
+        Parameters
+        ----------
+        sacc_data : sacc.Sacc
+            The data in the sacc format.
+        """
+        tracer = sacc_data.get_tracer(self.sacc_tracer)
+        beam = tracer.beam.copy().flatten()
```

Comment:
If you're gonna use the beam (and it doesn't seem like you are), just reading the beam doesn't tell you anything, since you don't know what ells it's sampled at.

Reply:
ahhh

Reply:
so I suggested removing the ell for the beam in an earlier comment because I thought it was something else

Reply:
let's remove the beam too then.

Reply:
Yeah I think so.